File: SiteVisitsNetCore2020Web/Models/PageTitleVariation.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.ComponentModel.DataAnnotations;

namespace SiteVisitsNetCore2020Web.Models
{
    public class PageTitleVariation
    {
        [Key]
        public string TitleVariation { get; set; }
        public string TitleVariationCompact { get; set; }
        public DateTime FirstOccurrence { get; set; }
        public virtual PageTitle PageTitle { get; set; }
    }
}

File: SiteVisitsNetCore2020Web/ClientApp/src/app/models/PageTitleVariation.ts
import { PageTitle } from './PageTitle';

export class PageTitleVariation {
    titleVariation: string;
    titleVariationCompact: string;
  pageTitle: PageTitle;
  constructor(options: {
    titleVariation: string,
    titleVariationCompact: string,
    pageTitle: PageTitle
  }) {
    this.titleVariation = options.titleVariation;
    this.titleVariationCompact = options.titleVariationCompact;
    this.pageTitle = new PageTitle(options.pageTitle);
  }

}
